### PR TITLE
Resolve "gitdist:  Create Table of SHA1s"

### PR DESCRIPTION
**Pull request by jmgate**
Here's an initial implementation that can satisfy #286.  Not saying it's a good one&mdash;mostly a copy, paste, and modify of the code for `gitdist dist-repo-status`.  No testing or documentation in place yet.  Just throwing this up here so we can iterate on it.

> **Warning:**  Do not merge this yet.  Lot's of clean-up needed.

### CODE REVIEW

**Comment by jmgate** [comment]
Right now this will generate the following
```
| Repo Dir | SHA1    | Commit Date         | Author                           | Summary                                                                              |
|:-------- |:-------:|:------------------- |:-------------------------------- |:------------------------------------------------------------------------------------ |
| TriBITS  | c0d7a33 | 2019-04-25 14:35:56 | jmgate@sandia.gov                | SAVEPOINT                                                                            |
| Trilinos | d55fd52 | 2019-04-25 08:03:13 | trilinos-autotester@trilinos.org | Merge Pull Request #5008 from bartlettroscoe/Trilinos/fix-seacas-install-permissions |
```

which will render as
| Repo Dir | SHA1    | Commit Date         | Author                           | Summary                                                                              |
|:-------- |:-------:|:------------------- |:-------------------------------- |:------------------------------------------------------------------------------------ |
| TriBITS  | c0d7a33 | 2019-04-25 14:35:56 | jmgate@sandia.gov                | SAVEPOINT                                                                            |
| Trilinos | d55fd52 | 2019-04-25 08:03:13 | trilinos-autotester@trilinos.org | Merge Pull Request #5008 from bartlettroscoe/Trilinos/fix-seacas-install-permissions |

I'm not really a fan of the rows in the table being multiple lines.  Possible solutions include:
1. Using a different date format.
1. Abbreviating the **Summary**.
1. Abbreviating the **Author**.
1. Omitting the **Commit Date**, **Author**, and **Summary** columns.  Perhaps with the inclusion of a `--short` option.

**Comment by jmgate** [comment]
Note that I do have an implementation (currently commented out) that abbreviates the **Summary**, which would generate the following
```
| Repo Dir | SHA1    | Commit Date         | Author                           | Summary                       |
|:-------- |:-------:|:------------------- |:-------------------------------- |:----------------------------- |
| TriBITS  | c0d7a33 | 2019-04-25 14:35:56 | jmgate@sandia.gov                | SAVEPOINT                     |
| Trilinos | d55fd52 | 2019-04-25 08:03:13 | trilinos-autotester@trilinos.org | Merge Pull Re...l-permissions |
```

rendering as

| Repo Dir | SHA1    | Commit Date         | Author                           | Summary                       |
|:-------- |:-------:|:------------------- |:-------------------------------- |:----------------------------- |
| TriBITS  | c0d7a33 | 2019-04-25 14:35:56 | jmgate@sandia.gov                | SAVEPOINT                     |
| Trilinos | d55fd52 | 2019-04-25 08:03:13 | trilinos-autotester@trilinos.org | Merge Pull Re...l-permissions |

In this case the **Summary** column is rather useless, because the majority of the time it'll say "Merge Pull...blah".

**Comment by jmgate** [comment]
Here's a possibility.  `gitdist dist-repo-version-table --dist-short` gives
```
| Repo Dir | SHA1    |
|:-------- |:-------:|
| TriBITS  | c0d7a33 |
| Trilinos | d55fd52 |
```

which renders as

| Repo Dir | SHA1    |
|:-------- |:-------:|
| TriBITS  | c0d7a33 |
| Trilinos | d55fd52 |

**Comment by bartlettroscoe** [comment]
FYI: Looks like causing an existing test failure shown [here](https://testing.sandia.gov/cdash/testDetails.php?test=76173138&build=4948034).

**Comment by jmgate** [comment]
Yeah I haven't messed with any of the testing yet.  May not get back to this till sometime next week though ☹️

**Comment by jmgate** [comment]
@bartlettroscoe, is the behavior shown in [this comment](https://github.com/TriBITSPub/TriBITS/pull/287#issuecomment-486832223) and [this comment](https://github.com/TriBITSPub/TriBITS/pull/287#issuecomment-486845427) acceptable to you?  I can work on testing and documentation if the behavior is where we want it.

**Comment by bartlettroscoe** [comment]
@jmgate asked:

> @bartlettroscoe, is the behavior shown in this comment and this comment acceptable to you? I can work on testing and documentation if the behavior is where we want it.

Yea, that looks pretty good to me.  Very nice!

**Comment by jmgate** [comment]
Sweet.  I'll start working on testing and documentation.  Probably won't have anything ready for review till next Friday.

**Comment by jmgate** [comment]
> Probably won't have anything ready for review till next Friday.

Or till I have time to hack on this in the midst of TUG 😁

**Comment by bartlettroscoe** [comment]
> @bartlettroscoe, does Travis CI still run on these PRs?

I had to disable Travis CI temporary since testing.sandia.gov/cdash/ is down and Travis CI can't submit to testing-dev.sandia.gov/cdash/.  But I could turn it back on and turn off the CDash submit.

**Comment by jmgate** [comment]
Oh, no worries.  I'm testing locally now.

File: test/python_utils/gitdist_UnitTests.py [hunk 1]
```diff
@@ -1507,6 +1641,37 @@ def test_dist_repo_status_all(self):
...
+        workingDir=testDir)
+      print(cmndOut.decode("ascii"))
+      cmndOut_expected = \
+        "| Repository     | SHA1    | Commit Date         | Author                 | Summary                                        |\n" \
+        "|:-------------- |:-------:|:------------------- |:---------------------- |:---------------------------------------------- |\n" \
+        "| MockProjectDir | e2dc488 | 2019-10-23 10:16:07 | user@domain.com        | Merge Pull Request #1234 from user/repo/branch |\n" \
+        "| ExtraRepo1     | f671414 | 2019-10-22 11:18:47 | wile.e.coyote@acme.com | Fixed a Bug                                    |\n" \
+        "| ExtraRepo2     | 50bbf3e | 2019-10-17 16:32:15 | someone@somwhere.com   | Did Some Work                                  |\n"
+      print(cmndOut_expected)
+      self.assertEqual(s(cmndOut), s(cmndOut_expected))
```
Comment (jmgate) [inline test/python_utils/gitdist_UnitTests.py]: @bartlettroscoe, can you help me with this bit here?  `cmndOut_expected` is
```
| Repository     | SHA1    | Commit Date         | Author                 | Summary                                        |
|:-------------- |:-------:|:------------------- |:---------------------- |:---------------------------------------------- |
| MockProjectDir | e2dc488 | 2019-10-23 10:16:07 | user@domain.com        | Merge Pull Request #1234 from user/repo/branch |
| ExtraRepo1     | f671414 | 2019-10-22 11:18:47 | wile.e.coyote@acme.com | Fixed a Bug                                    |
| ExtraRepo2     | 50bbf3e | 2019-10-17 16:32:15 | someone@somwhere.com   | Did Some Work                                  |
```

but `cmndOut` is
```
| Repository     | SHA1    | Commit Date | Author | Summary |
|:-------------- |:-------:|:----------- |:------ |:------- |
| MockProjectDir | e2dc488 |             |        |         |
| ExtraRepo1     | f671414 |             |        |         |
| ExtraRepo2     | 50bbf3e |             |        |         |
```

so it seems like the second, third, and fourth commands in `writeGitMockProgram_dist_repo_version_table()` aren't getting run.  I imagine you can quickly see where I've gone wrong here.  Haven't had much luck with `pdb` yet.  Thanks.

Comment (jmgate) [reply test/python_utils/gitdist_UnitTests.py]: Never mind.  I've figured out what's going on.  Should have a fix soon.

**Comment by jmgate** [comment]
@bartlettroscoe, I think this PR is finally good to go.  Squashed it down to a single commit and letting Travis CI do its thing.

**Comment by jmgate** [comment]
Looks like testing.sandia.gov might be hosed again.

**Comment by bartlettroscoe** [comment]
> Looks like testing.sandia.gov might be hosed again.

Yea, I just noticed that earlier today.  I will review the branch and test it locally.

**Comment by jmgate** [comment]
Oh bummer.  Let me take a look.

**Comment by jmgate** [comment]
Found the bug.  Now to cover it with a test 😁

**Comment by jmgate** [comment]
@bartlettroscoe, this is ready for review again.  You may want to make some changes to the `writeGitMockProgram*` routines.  I'm afraid I don't have a very good understanding of how `mockprogram.py` works, so it was just copy, paste, and modify until I got what I was expecting.

**Comment by jmgate** [comment]
Oh no worries.  I was glad I had time in TUG to work on it.  Thanks @bartlettroscoe

**Pull request merged**